tests/admission/test_jsonpatch.py

```diff
@@ -21,13 +32,33 @@ def test_removal_of_the_key():
 
 def test_addition_of_the_subkey():
     patch = Patch()
+    patch.original = {'xyz': {'def': 456}}
+    patch['xyz'] = {'abc': 123}
+    jsonpatch = patch.as_json_patch()
+    assert jsonpatch == [
+        {'op': 'add', 'path': '/xyz/abc', 'value': 123},
```

Comment:
Does it work, by the way? Or does it require `{op: add, path: /xyz, value: {}}` beforehand? I.e. the 2nd-level addition when the 1st level is absent — how is it interpreted by the webhook machinery from the Kubernetes side (I didn't try this case before)?

Reply:
In my limited testing with actual k8s resources (in this case, deployments), the single add works. If we need more detailed test cases in the real world, might have to create a CRD or something that be tested with more levels. Also, I am currently running against 1.17 clusters, which is considered rather old at this point, so have not checked the behaviour on newer versions.

I have tested with the following. The output of the JSONPatch is shown above each result.

(a) The annotation key already exists:
```
{'op': 'replace', 'path': '/spec/template/metadata/annotations/xyz3', 'value': 'abc3'}
---
  template:
    metadata:
      annotations:
        xyz3: abc3
```

(b) The annotation key does not exist, but `annotations` does.
```
 {'op': 'add', 'path': '/spec/template/metadata/annotations/xyz3', 'value': 'abc3'}
---
  template:
    metadata:
      annotations:
        something: else
        xyz3: abc3
```

(c) The key `annotations` does not exist at all.
```
{'op': 'add', 'path': '/spec/template/metadata/annotations', 'value': {'xyz2': 'abc2'}}
---
  template:
    metadata:
      annotations:
        xyz2: abc2
```

Reply:
Thanks for clarifying. That is what I wanted to know.

Speaking of annotations, I now wonder how it would handle annotations with "/" in them (e.g. `example.com/note: value`), and what is the supposed way of escaping slashes in JSONPatch. The internet says it is `~1` (and `~0` for the tilde itself). But this is a separate problem — let's not mix it here. It is better to do this as a separate PR.